cogs/mod.py

```diff
@@ -62,17 +62,23 @@ def check(r, u):
                     return await checkmsg.edit(content="Command timed out, canceling...")
        
             botmember = ctx.guild.me
-            if member.top_role > botmember.top_role:
-                return await ctx.send("My role is too low in the hierarchy. Please move it above the highest role the user you are trying to ban has.")
-            await ctx.message.delete()
-            messageok = f"You were banned from `{ctx.guild.name}` with reason:\n\n{reason}"
-            try:
-                await member.send(messageok)
-            except discord.errors.HTTPException:
-                pass
-            await member.ban(reason=f"Moderator: {ctx.message.author} | Reason: {reason}")
-            e = discord.Embed(title=f"{member} was banned | {reason}", color=config.red)
-            await ctx.send(embed=e)
+            if ctx.guild.get_member(ban_user.id) is not None:
+                member = ctx.guild.get_member(ban_user.id)
+                if member.top_role > botmember.top_role:
+                    return await ctx.send("My role is too low in the hierarchy. Please move it above the highest role the user you are trying to ban has.")
+                await ctx.message.delete()
+                messageok = f"You were banned from `{ctx.guild.name}` with reason:\n\n{reason}"
+                try:
+                    await member.send(messageok)
+                except discord.errors.HTTPException:
+                    pass
```

Comment:
There is no reason to have a a variable for the message here. It never gets used anywhere else.

Reply:
That's not me either.

Reply:
Could you still fix it, though? It's pretty simple to fix.

Reply:
Yes